src/main/java/com/lambdaworks/redis/cluster/ClusterTopologyRefresh.java

```diff
@@ -219,6 +219,10 @@ protected void close(Map<RedisURI, RedisAsyncConnectionImpl<String, String>> con
 
             try {
                 RedisAsyncConnectionImpl<String, String> connection = client.connectAsyncImpl(redisURI.getResolvedAddress());
+                char[] password = redisURI.getPassword();
+                if (password != null) {
```

Comment:
Could you also add a check if the password array length is not zero? `length != 0`